tests/slist.c

```diff
@@ -0,0 +1,21 @@
+#include "list.h"
+
+int main()
+{
+    struct slist head;
```

Comment:
This implementation has a big problem than you expect.
your head element is a stack variable while or others are on heap.
This is fine as long as you expect your function does not go out of scope and linker does not load another function at it's place.

Imagine, you are calling your library API "Y" from some function "X". Your head is a stack variable on "X". Down the line, list head is passed from function "Y"-->"Z"-->"A" for some hashing or lookup. Meanwhile, "X" is swapped out by loader. Address your list head is pointing to is now no longer available and "A" will crash and you will spend days finding out what did you wrong.

Always use heap.